### PR TITLE
Added fail_on_single_reject to prevent gateway failover

### PR DESCRIPTION
**Pull request by StevenGolemme**
This pull request is to resolve the issue to prevent gateway failover when callee rejects group confirm call.  Please see https://groups.google.com/forum/#!topic/2600hz-dev/AC_cXwSuZGQ for more details regarding this issue.

### CODE REVIEW

**Comment by fenollp** [comment]
@StevenGolemme Please rebase to master.

If 2600hz's remote is called `upstream` in your repo, try 

```shell
git fetch upstream && git merge upstream/master
```

**Comment by StevenGolemme** [comment]
@fenollp rebased.

File: applications/callflow/src/module/cf_device.erl [hunk 1]
```diff
@@ -55,7 +55,8 @@ bridge_to_endpoints(Data, Call) ->
     case cf_endpoint:build(EndpointId, Params, Call) of
         {'error', _}=E -> E;
         {'ok', Endpoints} ->
+            FailOnSingleReject = wh_json:get_value(<<"fail_on_single_reject">>, Data, 'undefined'),
             Timeout = wh_json:get_integer_value(<<"timeout">>, Data, ?DEFAULT_TIMEOUT_S),
             IgnoreEarlyMedia = cf_util:ignore_early_media(Endpoints),
-            whapps_call_command:b_bridge(Endpoints, Timeout, <<"simultaneous">>, IgnoreEarlyMedia, Call)
+            whapps_call_command:b_bridge(Endpoints, Timeout, <<"simultaneous">>, IgnoreEarlyMedia, 'undefined', 'undefined', 'undefined', FailOnSingleReject, Call)
```
Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_device.erl]: Hi @StevenGolemme,

First, thanks for the improvement. AFAICT this is a useful addition!

My only request is to build the bridge AMQP payload here now. The `whapps_call_command` functions were meant to be shortcuts to oft-used invocation. I think we've passed that point here.

Does that make sense?

Comment (StevenGolemme) [inline applications/callflow/src/module/cf_device.erl]: @jamesaimonetti  is this what you were thinking
```
Command = [{<<"Application-Name">>, <<"bridge">>}
                ,{<<"Endpoints">>, Endpoints}
                ,{<<"Timeout">>, Timeout}
                ,{<<"Ignore-Early-Media">>, IgnoreEarlyMedia}
                ,{<<"Fail-On-Single-Reject">>, FailOnSingleReject}
                ,{<<"Dial-Endpoint-Method">>, <<"simultaneous">>}
                ,{<<"Ignore-Forward">>, <<"false">>}
            ],
            whapps_call_command:send_command(Command, Call),
            whapps_call_command:b_bridge_wait(Timeout, Call)
```

Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_device.erl]: You got it

**Comment by k-anderson** [comment]
@StevenGolemme This is awesome!  Would you be willing to add the same parameter to cf_users and cf_ring_group for consistency?

**Comment by StevenGolemme** [comment]
@k-anderson cf_users and cf_ring_group should now be more consistent

**Pull request merged**